lib/broccoli/ember-app.js

```diff
@@ -1392,7 +1355,13 @@ class EmberApp {
   }
 
   test() {
-    let tests = this.addonPreprocessTree('test', this._processedTestsTree());
+    let addonTrees = this.addonTreesFor('test-support');
+    let mergedTests = mergeTrees(addonTrees.concat(this.trees.tests), {
+      overwrite: true,
+      annotation: 'TreeMerger (tests)',
+    });
+
+    let tests = this.addonPreprocessTree('test', this._defaultPackager.packageTests(mergedTests, this.name));
```

Comment:
I'd conceptually prefer to have the `packageTests` handle the full thing (e.g. moving `addonPreprocessTree`, `preprocessJs`, and `addonPostprocessTree`)...